packages/grafana-ui/src/components/Tabs/Tab.tsx

```diff
@@ -11,31 +11,44 @@ import { Counter } from './Counter';
 export interface TabProps extends HTMLProps<HTMLLIElement> {
   label: string;
   active?: boolean;
+  /** With this it is possible to use the tab as a hyperlink. Use in cases where the tabs update location. */
+  href?: string;
   icon?: IconName;
+  /** May cause unintended behaviour when used with href */
   onChangeTab: () => void;
+  /** A number next to the text. Usually used to display number of items in a tab's view. */
   counter?: number;
 }
 
 export const Tab = React.forwardRef<HTMLLIElement, TabProps>(
-  ({ label, active, icon, onChangeTab, counter, className, ...otherProps }, ref) => {
+  ({ label, active, icon, onChangeTab, counter, className, href, ...otherProps }, ref) => {
     const theme = useTheme();
     const tabsStyles = getTabStyles(theme);
+    const content = () => (
+      <>
+        {icon && <Icon name={icon} />}
+        {label}
+        {typeof counter === 'number' && <Counter value={counter} />}
+      </>
+    );
 
     return (
       <li
         {...otherProps}
-        className={cx(tabsStyles.tabItem, active && tabsStyles.activeStyle)}
-        onClick={() => {
-          if (!active) {
-            onChangeTab();
-          }
+        className={cx(!href && tabsStyles.padding, tabsStyles.tabItem, active && tabsStyles.activeStyle)}
+        onClick={(event: React.MouseEvent<HTMLLIElement>) => {
+          onChangeTab();
```

Comment:
Think this may be problematic. Imagine user providing a `href` and at the same time `onChangeTab`. He/she does not have a chance to cancel the event in that situation. We have two paths here:
1. If `href` is provided stop the propagation in the `onClick` handler.
2. Change signature of the onChangeTab to be `onChangeTab(event: ...)` so that the responsibility is delegated to the consumer.

Think the second approach is the one we want. For example ant.d follows this pattern: https://ant.design/components/tabs/